components/bpmn/org.wso2.carbon.bpmn.analytics.publisher/src/main/java/org/wso2/carbon/bpmn/analytics/publisher/AnalyticsPublisher.java

```diff
@@ -54,31 +56,38 @@
-                int tenantId = CarbonContext.getThreadLocalCarbonContext().getTenantId();
-                String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
-                Registry registry = BPMNAnalyticsHolder.getInstance().getRegistryService()
-                        .getGovernanceSystemRegistry();
-                if (processInstanceStreamId == null && taskInstanceStreamId == null) {
-                    if (log.isDebugEnabled()) {
-                        log.debug("Stream definitions are null...");
+            if (BPMNDataReceiverConfig.isDASPublisherActivated()) {
+                RegistryUtils.setTrustStoreSystemProperties();
+                dataPublisher = createDataPublisher();
+                if (dataPublisher != null) {
+                    processInstanceStreamId = getBPMNProcessInstanceStreamId();
+                    taskInstanceStreamId = getBPMNTaskInstanceStreamId();
+                    analyticsPublishServiceUtils = new AnalyticsPublishServiceUtils();
+                    int tenantId = CarbonContext.getThreadLocalCarbonContext().getTenantId();
+                    String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+                    Registry registry = BPMNAnalyticsHolder.getInstance().getRegistryService()
+                            .getGovernanceSystemRegistry();
+                    if (processInstanceStreamId == null && taskInstanceStreamId == null) {
+                        if (log.isDebugEnabled()) {
+                            log.debug("Stream definitions are null...");
+                        }
+                    } else {
+                        setPrivilegeContext(tenantId, tenantDomain, registry);
                     }
                 } else {
-                    setPrivilegeContext(tenantId, tenantDomain, registry);
+                    if (log.isDebugEnabled()) {
+                        log.debug("Data Publisher object is null...");
+                    }
+                }
+            } else {
+                if (log.isDebugEnabled()) {
+                    log.debug("BPMN Data Publisher is not activated...");
                 }
-            }else{
-             if(log.isDebugEnabled()){
-                 log.debug("Data Publisher object is null...");
-             }
             }
-        } catch (MalformedURLException | AgentException | AuthenticationException | TransportException |
+        } catch (IOException | AgentException | AuthenticationException | TransportException |
                 DifferentStreamDefinitionAlreadyDefinedException | StreamDefinitionException |
-                MalformedStreamDefinitionException | UserStoreException | RegistryException e) {
+                MalformedStreamDefinitionException | UserStoreException | RegistryException | XMLStreamException
+                e) {
             String errMsg = "Data publisher objects initialization error.";
             log.error(errMsg, e);
```

Comment:
If errMsg is not used anywhere else, use the literal string directly in log.error instead of using a separate variable